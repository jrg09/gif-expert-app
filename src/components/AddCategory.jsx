import { useRef, useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    const newValue = inputValue.trim();

    if (newValue.length <= 1) {
      return;
    }

    onNewCategory(newValue);

    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit} aria-label="formulario">
      <input
        type="text"
        value={inputValue}
        onChange={handleChange}
        placeholder="Buscar GIFs"
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
