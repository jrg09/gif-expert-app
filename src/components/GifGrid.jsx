import { useFetchGifs, GifItem } from "./";
import { InfinitySpin } from "react-loader-spinner";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  //console.log({ gifs, isLoading });

  return (
    <div>
      <h4>{category}</h4>
      {isLoading ? (
        <div>
          <InfinitySpin width="200" color="#4fa94d" />
        </div>
      ) : (
        <>
          <h5>Total gifs: {gifs.length}</h5>
          <div className="card-grid">
            {gifs.map((gif) => (
              <GifItem key={gif.id} {...gif} />
            ))}
          </div>
        </>
      )}
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
