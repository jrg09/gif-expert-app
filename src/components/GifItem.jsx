import PropTypes from "prop-types";

export const GifItem = ({ id, title, image }) => {
  return (
    <>
      <div className="card" style={{ width: "10rem" }}>
        <img src={image} alt={title} />
        <p className="small">{title}</p>
      </div>
    </>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
