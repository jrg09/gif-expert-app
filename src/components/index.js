//archivo de barril, archivo Ã­ndice
export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifItem";
export * from "../hooks/useFetchGifs";
