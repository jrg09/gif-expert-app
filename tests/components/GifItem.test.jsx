import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const title = "Goku";
  const image = "https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/100w_s.gif";

  test("01 pruebas de PropTypes obligatorios y hacer match con snapshot doce", () => {
    const { container } = render(<GifItem title={title} image={image} />);
    //screen.debug();

    expect(container).toMatchSnapshot();
  });

  test("02 should mostrar imagen con imagen y alt enviados", () => {
    render(<GifItem title={title} image={image} />);
    //screen.debug();
    //console.log('    screen.getByRole("img").src', screen.getByRole("img").src);

    //expect(screen.getByRole("img").src).toBe(image);
    //expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect({ src, alt }).toStrictEqual({ src: image, alt: title });
  });

  test("03 should mostrar titulo en el componente", () => {
    render(<GifItem title={title} image={image} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
