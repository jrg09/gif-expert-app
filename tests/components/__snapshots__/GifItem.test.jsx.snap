// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem 01 pruebas de PropTypes obligatorios y hacer match con snapshot doce 1`] = `
<div>
  <div
    class="card"
    style="width: 10rem;"
  >
    <img
      alt="Goku"
      src="https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/100w_s.gif"
    />
    <p
      class="small"
    >
      Goku
    </p>
  </div>
</div>
`;
